govcd/vapp.go

```diff
@@ -772,27 +772,50 @@ func (vapp *VApp) ChangeVMName(name string) (Task, error) {
 
 }
 
-func (vapp *VApp) DeleteMetadata(key string) (Task, error) {
-	err := vapp.Refresh()
+func (vapp *VApp) GetMetadata(requestUri string) (*types.Metadata, error) {
+	return getMetadata(vapp.client, requestUri)
+}
+
+func getMetadata(client *Client, requestUri string) (*types.Metadata, error) {
+	metadata := &types.Metadata{}
+
+	getMetadata, _ := url.ParseRequestURI(requestUri + "/metadata/")
+
+	req := client.NewRequest(map[string]string{}, "GET", *getMetadata, nil)
+
+	req.Header.Add("Content-Type", "application/vnd.vmware.vcloud.metadata+xml")
+
+	resp, err := checkResp(client.Http.Do(req))
 	if err != nil {
-		return Task{}, fmt.Errorf("error refreshing vApp before running customization: %v", err)
+		return metadata, fmt.Errorf("error retrieving task: %s", err)
 	}
 
-	if vapp.VApp.Children == nil {
-		return Task{}, fmt.Errorf("vApp doesn't contain any children, aborting customization")
+	if err = decodeBody(resp, metadata); err != nil {
+		return metadata, fmt.Errorf("error decoding task response: %s", err)
 	}
 
-	apiEndpoint, _ := url.ParseRequestURI(vapp.VApp.Children.VM[0].HREF)
+	// The request was successful
+	return metadata, nil
+}
+
+func (vapp *VApp) DeleteMetadata(key string) (Task, error) {
```

Comment:
Please add an SDK-user targeted comment (documentation) about this public function.

Reply:
Fixed.